pants-plugins/internal_plugins/versioned_setup/register.py

```diff
@@ -0,0 +1,81 @@
+import os.path
+
+from pants.backend.python.goals.setup_py import SetupKwargs, SetupKwargsRequest
+from pants.engine.fs import DigestContents, GlobMatchErrorBehavior, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+
+
+class VersionedSetupKwargsRequest(SetupKwargsRequest):
+    @classmethod
+    def is_applicable(cls, _: Target) -> bool:
+        # We always use our custom `setup()` kwargs generator for `python_distribution`
```

Comment:
which custom `setup` is this @kilemensi  ?

Reply:
[This](https://github.com/CodeForAfrica/api/blob/75bba6d1311d4384af435b051734086b42de0cb4/dj-cmd/py/BUILD#L17) @esirK

Reply:
ooh okay. Thought it meant we have a `setup` function